aws-datastore/src/main/java/com/amplifyframework/datastore/storage/StorageItemChange.java

```diff
@@ -46,9 +49,20 @@ private StorageItemChange(
             @NonNull Type type,
             @NonNull T item,
             @NonNull Class<T> itemClass) {
+        this(changeId, initiator, type, null, item, itemClass);
+    }
+
+    private StorageItemChange(
+            @NonNull UUID changeId,
+            @NonNull Initiator initiator,
+            @NonNull Type type,
+            @Nullable QueryPredicate predicate,
```

Comment:
Alternately, instead of using `null` to mean "a predicate which doesn't check anything", we can have a

```
// A factory class to build some different types of query predicates.
final class QueryPredicates {
    private QueryPredicates() {}

    // Builds a query predicate that doesn't filter out anything.
    static QueryPredicate matchAll() {
        return new QueryPredicate() {}
    }
}
```

And then pass in a:
```
QueryPredicates.matchAll()
```

Reply:
This is a good idea, but `QueryPredicate` interface/class is sensitive in that GraphQL implementation also relies on it, so I think refactoring it should involve a separate PR with David's input as well.